### PR TITLE
Constellation Lab Lighthouse - Widget - Case Launcher

### PR DESCRIPTION
**Pull request by iglep**
This pull request is to provide access to the Case Launcher (Widget) DX Component to be reviewed.

### CODE REVIEW

File: src/components/Pega_Extensions_CaseLauncher/styles.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// utilizing theming, comment out, if want individual style
+import styled, { css } from 'styled-components';
+import { Configuration, defaultThemeProp } from '@pega/cosmos-react-core';
+
+export default styled(Configuration)``;
```
Comment (ricmars) [inline src/components/Pega_Extensions_CaseLauncher/styles.ts:5]: move Configuration to the index.js - We avoid in this repo doing that in the styles.ts - see other components

File: src/components/Pega_Extensions_CaseLauncher/styles.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+// individual style, comment out above, and uncomment here and add styles
```
Comment (ricmars) [inline src/components/Pega_Extensions_CaseLauncher/styles.ts:7]: remove commented code

**Comment by iglep** [comment]
I submitted a new commit.

File: src/components/Pega_Extensions_CaseLauncher/index.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  /** Description of the case launched by the widget */
+  description: string;
+  /** Class name of the case; used as property when launching pyStartCase  */
+  classFilter: any;
+  /** Primary button label  */
+  labelPrimaryButton: string;
+  getPConnect: any;
+};
+
+// Duplicated runtime code from Constellation Design System Component
```
Comment (ricmars) [inline src/components/Pega_Extensions_CaseLauncher/index.tsx:24]: remove this comment

File: src/components/Pega_Extensions_CaseLauncher/index.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      containerName: 'primary',
+      openCaseViewAfterCreate: true
+    };
+
+    pConn.getActionsApi().createWork(className, options);
+  };
+
+  return (
+    <Configuration>
+      <Card>
```
Comment (ricmars) [inline src/components/Pega_Extensions_CaseLauncher/index.tsx:45]: missing your styled component as 'XXX'

File: src/components/Pega_Extensions_CaseLauncher/styles.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// utilizing theming, comment out, if want individual style
+import styled, { css } from 'styled-components';
+import { type themeDefinition } from '@pega/cosmos-react-core';
+
+export const StyledCard = styled.article(({ theme }: { theme: typeof themeDefinition }) => {
```
Comment (ricmars) [inline src/components/Pega_Extensions_CaseLauncher/styles.ts:5]: should be export default styled.article... - there should be a validation error

File: src/components/Pega_Extensions_CaseLauncher/index.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  classFilter: any;
+  /** Primary button label  */
+  labelPrimaryButton: string;
+  getPConnect: any;
+};
+
+// Duplicated runtime code from Constellation Design System Component
+
+// props passed in combination of props from property panel (config.json) and run time props from Constellation
+// any default values in config.pros should be set in defaultProps at bottom of this file
```
Comment (ricmars) [inline src/components/Pega_Extensions_CaseLauncher/index.tsx:27]: remove this comment - this is not true

File: src/components/Pega_Extensions_CaseLauncher/index.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+export default function PegaExtensionsCaseLauncher(props: CaseLauncherProps) {
+  const { heading, description, classFilter, labelPrimaryButton, getPConnect } = props;
+  const pConn = getPConnect();
+
+  /* Create a new case on click of the selected button */
+  const createCase = (className: any) => {
```
Comment (ricmars) [inline src/components/Pega_Extensions_CaseLauncher/index.tsx:33]: should be string type

File: src/components/Pega_Extensions_CaseLauncher/index.tsx (hunk 4 above)
Comment (iglep) [reply src/components/Pega_Extensions_CaseLauncher/index.tsx:45]: I'm using the StyledCard imported from styles.ts ... If I export default the "styled.article" the declaration using as 'XXX' , will I need to use this that way?

Comment (ricmars) [reply src/components/Pega_Extensions_CaseLauncher/index.tsx:45]: sorry XXX was just meant to be a word

Comment (ricmars) [reply src/components/Pega_Extensions_CaseLauncher/index.tsx:45]: you do import MainCardContent from './styles'

Comment (ricmars) [reply src/components/Pega_Extensions_CaseLauncher/index.tsx:45]: look at the Banner styles.ts

File: src/components/Pega_Extensions_CaseLauncher/styles.ts (hunk 5 above)
Comment (ricmars) [reply src/components/Pega_Extensions_CaseLauncher/styles.ts:5]: look at banner styles.ts

**Pull request merged**